### PR TITLE
uadk bugfix and optimize

### PR DESCRIPTION
**Pull request by lin755**

### CODE REVIEW

File: wd_rsa.c [hunk 1]
```diff
@@ -836,6 +836,11 @@ handle_t wd_rsa_alloc_sess(struct wd_rsa_sess_setup *setup)
 	/* Some simple scheduler don't need scheduling parameters */
 	sess->sched_key = (void *)wd_rsa_setting.sched.sched_init(
 		     wd_rsa_setting.sched.h_sched_ctx, setup->sched_param);
+	if (WD_IS_ERR(sess->sched_key)) {
+		WD_ERR("failed to init session schedule key!\n");
+		free(sess);
```
Comment (lin755) [inline wd_rsa.c:841]: mem leak here

Comment (lin755) [reply wd_rsa.c]: fixed

**Review by Liulongfang — COMMENTED** [review]
OK

**Review by Liulongfang — COMMENTED** [review]
complete review, looks good!

**Pull request merged**